mpifileutils.spec

```
@@ -44,14 +44,15 @@
 
 Name:		mpifileutils
 Version:	0.11
-Release:	1%{?relval}%{?dist}
+Release:	2%{?relval}%{?dist}
 Summary:	File utilities designed for scalability and performance.
 
 Group:		System Environment/Libraries
 License:	Copyright and BSD License
 URL:		https://hpc.github.io/mpifileutils
 Source:		https://github.com/hpc/%{name}/archive/v%{version}.tar.gz
-BuildRoot:  %_topdir/BUILDROOT
+Patch1:		v%{version}..%{?git_short}.patch
```

You cannot **require** the setting of macro values external to the specfile.  You can have it optionally allowed, but the user should not **have** to supply anything to rpmbuild externally.  So maybe make the patch application dependent on the external specification, otherwise just build the non-patched version:

```suggestion
%if defined(git_short)
Patch1:		v%{version}..%{?git_short}.patch
%endif
```

Please test that, as my memory of `%defined()` is hazy.

---

It seems that `%if 0%{?git_short:1}` should work, per https://backreference.org/2011/09/17/some-tips-on-rpm-conditional-macros/

---

> It seems that `%if 0%{?git_short:1}` should work, per https://backreference.org/2011/09/17/some-tips-on-rpm-conditional-macros/

Indeed.  You can see an explanation on that same page of `%defined()` if you search for _More elaborate variation of conditional expansion can be found in the official rpm documentation. For example, they use some sort of "function-like" macros as follows_.

`%defined()` is just providing a more clear functional value of that.

---

I don't know what my brain was thinking...
You're right, this works: `%if %{defined git_short}`
Sorry about that